packages/expo-file-system/ios/EXFileSystem/EXSessionTasks/EXSessionTaskDispatcher.h

```diff
@@ -0,0 +1,22 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+#import <Foundation/Foundation.h>
+#import <EXFileSystem/EXSessionTaskDelegate.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface EXSessionTaskDispatcher : NSObject <NSURLSessionDelegate>
+
+- (void)registerTaskDelegate:(EXSessionTaskDelegate *)delegate forTask:(NSURLSessionTask *)task;
+
+- (void)registerResumableDownloadTask:(NSURLSessionDownloadTask *)task uuid:(NSString *)uuid;
```

Comment:
I'm afraid we're putting too much responsibility into one class — do you think it would make sense to extract the resumable downloads managing feature to a separate class? This would make the dispatcher much more generic (only dispatching events to tasks) and the resumable tasks holder somewhere else (I imagine we could pass the `EXResumablesManager` as a delegate to `EXSessionTaskDelegate`)

![excalidraw-202012311929-32](https://user-images.githubusercontent.com/1151041/79329285-ce47aa80-7f17-11ea-84a3-1d830a9706d6.png)

What do you think?